### ISSUE TITLE
[a11y] Ensures links have discernible text

### ISSUE THREAD

**Issue by pdehaan** [mozilla/fxa#12869]
## Description

I was goofin' around with `@axe-core/playwright` and it found one issue so far on the homepage (and /reset_password page, but it's the same offending element).

Looks like the "about-mozilla" logo in the bottom left is missing some a11y attributes.

### Steps to reproduce

This is still a bit early, but the idea was that we could wire this into the existing functional-test e2e pipeline for some FxA+SubPlat tests. I'm a bit limited as to which pages I can scape without logging in, but I know @ashrivastava-qa already has some tests that she did for SubPlat which we can copy-pasta and then figure out if these should run on local/stage/prod/all.

```js
// import { inspect } from "node:util";

import dedent from "dedent";
import AxeBuilder from "@axe-core/playwright";
import playwright from "playwright";

const BASE_URL = "https://accounts.stage.mozaws.net";
const CODE_FENCE = "```";
const DISABLE_RULES = [
  "color-contrast",
  "landmark-one-main",
  "meta-viewport-large",
  "region",
];

const browser = await playwright.firefox.launch();
const context = await browser.newContext();
const page = await context.newPage();

await a11yLint(page, "/signin");
await a11yLint(page, "/reset_password");

async function a11yLint(page, href) {
  // Make sure we absoute URL this...
  href = new URL(href, BASE_URL).href;
  await page.goto(href);
  try {
    const { violations } = await new AxeBuilder.default({ page })
      .disableRules(DISABLE_RULES)
      .analyze();

    console.log(`\n---\nFound ${violations.length} violation(s) on ${href}`);
    for (const violation of violations) {
      console.log(`\n[${violation.impact}] ${violation.description}`);
      for (const node of violation.nodes) {
        console.log(dedent`
          ${CODE_FENCE}
          ${node.html}
          ${CODE_FENCE}
        `);
        console.log("\n" + node.failureSummary);
      }
    }
    // console.log(inspect(violations, { depth: 5 }));
  } catch (err) {
    console.error("error");
    console.error(err);
    process.exitCode = 1;
  }
}

await browser.close();
```

### Expected result

No errors.

### Actual result

This snippet is missing some `aria-label` or `title` attributes or any usable information for people with a screen reader.

```html
<a id="about-mozilla" rel="author" target="_blank" href="https://www.mozilla.org/about/?utm_source=firefox-accounts&amp;utm_medium=Referral"></a>
```

````
Found 1 violation(s) on https://accounts.stage.mozaws.net/signin

[serious] Ensures links have discernible text
```
<a id="about-mozilla" rel="author" target="_blank" href="https://www.mozilla.org/about/?utm_source=firefox-accounts&amp;utm_medium=Referral"></a>
```

Fix all of the following:
  Element is in tab order and does not have accessible text

Fix any of the following:
  Element does not have text that is visible to screen readers
  aria-label attribute does not exist or is empty
  aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
  Element has no title attribute
````

````
Found 1 violation(s) on https://accounts.stage.mozaws.net/reset_password

[serious] Ensures links have discernible text
```
<a id="about-mozilla" rel="author" target="_blank" href="https://www.mozilla.org/about/?utm_source=firefox-accounts&amp;utm_medium=Referral"></a>
```

Fix all of the following:
  Element is in tab order and does not have accessible text

Fix any of the following:
  Element does not have text that is visible to screen readers
  aria-label attribute does not exist or is empty
  aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
  Element has no title attribute
````

### Environment

What browser and operating system are you using?
